lib/ferrum/cookies.rb

```diff
@@ -61,10 +63,11 @@ def [](name)
       all[name]
     end
 
-    def set(name: nil, value: nil, **options)
-      cookie = options.dup
-      cookie[:name]   ||= name
-      cookie[:value]  ||= value
+    def set(options)
+      cookie = (
+        options.is_a?(Cookie) ? options.attributes : options
+      ).dup.transform_keys(&:to_sym)
```

Comment:
`transform_keys` looks redundant
```suggestion
      ).dup
```

Reply:
@Mifrill Thanks for the suggestion!

I did some experiments, and while this looks redundant, it's needed and makes me realize that I'm not testing the feature enough.

The stored attributes will use string keys when cookies are defined through the browser. So, the following use cases would fail:
```ruby
browser.cookies.set(cookie)

browser.cookies.set(cookie.attributes)
```

For example, the `domain` would be overwritten by `cookie[:domain] ||= default_domain` as `cookie[:domain]` != `cookie["domain"]`.

I updated the PR, including more tests, to make it more explicit and also ensure the following expected behaviors:

- We do not create side effects by mutating the options Hash: [L136](https://github.com/rubycdp/ferrum/pull/256/files#diff-a880348c9c62b3e07cb738dd7539f6a10a9ebd0a2af872c9db2d84777800791eR136)
- We do not create side effects by mutating the cookie Object. [146](https://github.com/rubycdp/ferrum/pull/256/files#diff-a880348c9c62b3e07cb738dd7539f6a10a9ebd0a2af872c9db2d84777800791eR146)
- We ensure all the characteristics of the reference cookie. [L106](https://github.com/rubycdp/ferrum/pull/256/files#diff-a880348c9c62b3e07cb738dd7539f6a10a9ebd0a2af872c9db2d84777800791eR106)
- We ensure all the characteristics of the reference attributes: [L129](https://github.com/rubycdp/ferrum/pull/256/files#diff-a880348c9c62b3e07cb738dd7539f6a10a9ebd0a2af872c9db2d84777800791eR129)

That way, if we remove either `dup` or `transform_keys(&:to_sym)`, the tests will fail.

Reply:
alright, thanks for thorough explanation :+1: